doc/api/metrics.md

```diff
@@ -0,0 +1,1040 @@
+import { uniqueSet } from 'node:metrics';
+
+const uniqueUsers = uniqueSet('users.unique');
+
+uniqueUsers.add(user);        // Only counted once per unique value
+uniqueUsers.add(user);        // Ignored because user is already in the set
+uniqueUsers.add(anotherUser, { source: 'login' }); // Count another user with metadata
+```
+
+#### `uniqueSet.count`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The number of unique values seen.
+
+### Class: `PeriodicGauge`
+
+* Extends: {metrics.Gauge}
+
+<!-- YAML
+added: REPLACEME
+-->
+
+A gauge that automatically samples values at regular intervals.
+
+#### `periodicGauge.metric`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {metrics.Metric}
+
+The underlying metric instance used for reporting.
+
+#### `periodicGauge.interval`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The sampling interval in milliseconds. Setting this property reschedules the timer.
+
+#### `periodicGauge.schedule()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+Schedules the periodic sampling based on the configured interval. This is called
+automatically when the gauge is created, but can be called again to reschedule
+after it has been stopped.
+
+```mjs
+import { periodicGauge } from 'node:metrics';
+import { cpuUsage } from 'node:process';
+
+const cpu = periodicGauge('cpu.usage', 5000, () => {
+  return cpuUsage().user;
+});
+
+cpu.stop();
+
+// Reschedule sampling
+cpu.schedule();
+```
+
+#### `periodicGauge.stop()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+Stops the periodic sampling.
+
+```mjs
+import { periodicGauge } from 'node:metrics';
+import { cpuUsage } from 'node:process';
+
+const cpu = periodicGauge('cpu.usage', 5000, () => {
+  return cpuUsage().user;
+});
+
+// Stop sampling when no longer needed
+cpu.stop();
+```
+
+#### `periodicGauge[Symbol.dispose]()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+Allows `using` syntax to automatically stop the periodic gauge when done.
+
+```mjs
+import { periodicGauge } from 'node:metrics';
+import { cpuUsage } from 'node:process';
+
+{
+  using cpu = periodicGauge('cpu.usage', 1000, () => {
+    return cpuUsage().user;
+  });
+
+  // Perform operations that require periodic sampling...
+
+  // Sampling is automatically stopped here
+}
+```
+
+## Metric Streams
+
+### `metrics.statsdStream()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {stream.Transform}
+
+Creates a transform stream that converts all metrics to StatsD format.
+
+```mjs
+import { statsdStream } from 'node:metrics';
+import { stdout } from 'node:process';
+
+const stream = statsdStream();
+stream.pipe(stdout);
+```
+
+Output format: `metric.name:value|type`
+
+### `metrics.dogstatsdStream()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {stream.Transform}
+
+Creates a transform stream that converts all metrics to DogStatsD format with tags.
+
+```mjs
+import { dogstatsdStream } from 'node:metrics';
+import { stdout } from 'node:process';
+
+const stream = dogstatsdStream();
+stream.pipe(stdout);
```

Comment:
It doesn't need to be figured out in the initial version here, but it would be good to document examples of filtering these streams or passing them through transforms. Also, how are you imagining sampling to be implemented?